File: easy/The Descent/PHP/The Descent.php
<?php
/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/


// game loop
while (TRUE)
{
    $biggest = 0;
    $num = 0;

    for ($i = 0; $i < 8; $i++)
    {
        // $mountainH: represents the height of one mountain.
        fscanf(STDIN, "%d", $mountainH);

        if ($biggest < $mountainH){
            $biggest = $mountainH;
            $num = $i;
        }
    }

    // Write an action using echo(). DON'T FORGET THE TRAILING \n
    // To debug: error_log(var_export($var, true)); (equivalent to var_dump)

    echo("$num\n"); // The index of the mountain to fire on.
}
?>

File: easy/The Descent/Haskell/The Descent.hs
import System.IO
import Control.Monad
import Data.IORef

main :: IO ()
main = do
    hSetBuffering stdout NoBuffering -- DO NOT REMOVE
    
    -- The while loop represents the game.
    -- Each iteration represents a turn of the game
    -- where you are given inputs (the heights of the mountains)
    -- and where you have to print an output (the index of the mountain to fire on)
    -- The inputs you are given are automatically updated according to your last actions.
    
    
    -- game loop
    forever $ do
        biggest <- newIORef 0
        num <- newIORef 0
        
        forM_ [0..7] $ \i -> do
            input_line <- getLine
            let mountainh = read input_line :: Int -- represents the height of one mountain.
            
            biggestVal <- readIORef biggest
            when (mountainh > biggestVal) $ do
                writeIORef biggest mountainh
                writeIORef num i
            
        
        -- hPutStrLn stderr "Debug messages..."
        
        -- The index of the mountain to fire on.
        result <- readIORef num
        print result